caching-big-repositories_5bf03fb92c7d3a31944e139d.md

```diff
@@ -0,0 +1,105 @@
+
+* [Overview](#overview)
+* [The logic behind the recipe](#the-logic-behind-the-recipe)
+* [An example project](#an-example-project)
+* [See Also](#see-also)
+
+## Overview
+
+Imagine that you have one or multiple large files on a GitHub repository and
```

Comment:
This is technically not fully correct, eg more than half of a Git repo's size can be in history of past files, some of which were big. Anyway it is beyond scope of this guide to discuss why a Git repo can be large. It needs to start from the fact that somebody just has a large Git repo.

So we can streamline the intro by saying something like:

_This guide shows you how to cache a large Git repository and avoid cloning it from scratch in every block._

Reply:
Done.